### PR TITLE
index ip field

### PR DESCRIPTION
**Pull request by PSeitz**
- refactor Term
- fix truncate, remove mutable access from term
- remove truncate
- add indexing for ip field

### CODE REVIEW

File: src/query/query_parser/query_parser.rs [hunk 1]
```diff
@@ -401,7 +405,7 @@ impl QueryParser {
                 Ok(Term::from_field_bytes(field, &bytes))
             }
             FieldType::IpAddr(_) => Err(QueryParserError::UnsupportedQuery(
-                "Range query are not supported on IpAddr field.".to_string(),
+                "Range query are not supported on ip field.".to_string(),
```
Comment (fulmicoton) [inline src/query/query_parser/query_parser.rs:408]: Is there an issue open for this?

Comment (PSeitz) [reply src/query/query_parser/query_parser.rs:408]: https://github.com/quickwit-oss/tantivy/issues/1626

File: src/schema/ip_options.rs [hunk 2]
```diff
@@ -79,6 +134,8 @@ impl From<StoredFlag> for IpAddrOptions {
 impl From<IndexedFlag> for IpAddrOptions {
     fn from(_: IndexedFlag) -> Self {
         IpAddrOptions {
+            fieldnorms: true,
```
Comment (fulmicoton) [inline src/schema/ip_options.rs:137]: maybe we can drop fieldnorms for IP addresses?
what do you think?

**Review by fulmicoton — APPROVED** [review]

File: src/schema/ip_options.rs (hunk 2 above)
Comment (PSeitz) [reply src/schema/ip_options.rs:137]: I would keep it to be able to use BM25

**Pull request merged**